assignment1/src/main/java/ro/utcn/sd/he/assignment1/persistence/jdbc/JdbcQuestionRepository.java

```diff
@@ -0,0 +1,77 @@
+package ro.utcn.sd.he.assignment1.persistence.jdbc;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import ro.utcn.sd.he.assignment1.model.Question;
+import ro.utcn.sd.he.assignment1.persistence.api.QuestionRepository;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+public class JdbcQuestionRepository implements QuestionRepository {
+    private final JdbcTemplate template;
+
+
+    @Override
+    public Question save(Question question) {
+        if(question.getId() != 0){
+            update(question);
+        }else {
+            int id = insert(question);
+            question.setId(id);
+        }
+        return question;
+    }
+
+    @Override
+    public Optional<Question> findById(int id) {
+        List<Question> questions = template.query("SELECT * FROM question WHERE id = ?",
+                    new Object[] {id},
+                (resultSet, i) ->  new Question(resultSet.getInt("id"),
+                        resultSet.getString("author"),
+                        resultSet.getString("title"),
+                        resultSet.getString("text"),
+                        resultSet.getTimestamp("creation_date_time")));
+
+
+        return questions.isEmpty() ? Optional.empty() : Optional.of(questions.get(0));
+    }
+
+    @Override
+    public void remove(Question question) {
+        template.update("DELETE FROM question WHERE id = ?", question.getId());
+    }
+
+    @Override
+    public List<Question> findAll() {
+
+        return template.query("SELECT * FROM question", (resultSet, i ) ->
+                new Question(resultSet.getInt("id"),
```

Comment:
You could extract this RowMapper into a separate class instead of duplicating the lambda.